### PR TITLE
session and transaction doc

### PR DESCRIPTION
**Pull request by mmeloni**

### CODE REVIEW

File: src/master/develop/transactions.md [hunk 1]
```diff
@@ -1,6 +1,89 @@
 # Transactions
+Immudb supports transactions both on key-value and SQL level, but interactive transactions are supported only on SQL with the exception of `execAll` method, that provides some additional properties.
 
-`GetAll`, `SetAll` and `ExecAll` are the foundation of transactions in immudb. They allow the execution of a group of commands in a single step, with two important guarantees:
+## SQL interactive transactions
+Interactive transactions are a way to execute multiple SQL statements in a single transaction having the possibility to mix application logic with SQL statements.
```
Comment (zaza81) [inline src/master/develop/transactions.md:5]: transaction. This makes possible to delegate application logic to SQL statements - a very common use case is for example checking if the balance > 0 before making a purchase.

**Review by zaza81 — APPROVED** [review]
I just suggest a rephrasing

File: src/master/develop/connection.md [hunk 2]
```diff
@@ -84,9 +86,9 @@ This generates a list of folders containing certificates and private keys to set
 
 ::: tab Go
 ```go
-	client, err := c.NewImmuClient(
-		c.DefaultOptions().WithMTLsOptions(
-			c.MTLsOptions{}.WithCertificate("{path-to-immudb-src-folder}/tools/mtls/4_client/certs/localhost.cert.pem").
+	c := immudb.NewClient().WithOptions(
```
Comment (byo) [inline src/master/develop/connection.md:89]: Something's wrong with code indentation here?

File: src/master/develop/transactions.md [hunk 3]
```diff
@@ -1,6 +1,89 @@
...
+```
+Commit() CommittedSQLTx, error
+Rollback() error
+SQLExec(sql, params) error
+SQLQuery(sql, params) SQLQueryResult, error
+```
+It's possible to rollback a transaction by calling the `Rollback()` method. In this case, the transaction object is no longer valid and should not be used anymore.
+To commit a transaction, you must call the `Commit()` method.
+
+> **Note**: At the moment immudb support only 1 read-write transaction at a time, so it's up the application to ensure that only one read-write transaction is open at a time, or to handle read coflict error.
```
Comment (byo) [inline src/master/develop/transactions.md:17]: TODO: What would be the result of a transaction conflict? Can we have some snippet that shows how to determine if there's a conflict?

File: src/master/develop/connection.md [hunk 4]
```diff
@@ -7,13 +7,15 @@ You can modify defaults on the immudb server in `immudb.toml` in the config fold
...
-}
-ctx := context.Background()
-// login with default username and password
-_ , err = c.Login(ctx, []byte(`immudb`), []byte(`immudb`))
+import (
+  "context"
+  immudb "github.com/codenotary/immudb/pkg/client"
+  "log"
+)
+```
```
Comment (byo) [inline src/master/develop/connection.md:15]: Minor: There are some markdown linting errors in this file, if using vscode, this one is a pretty nice tool to find all issues: https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

File: src/master/jumpstart.md [hunk 5]
```diff
@@ -340,19 +338,23 @@ In order to use SQL from the Go SDK, you create a immudb client and login to the
...
-Then you can create the client and login to the database:
+Then you can create the client and open a new session to the database:
 
 ```go
-  c, err := client.NewImmuClient(client.DefaultOptions())
-	if err != nil {
-		log.Fatal(err)
-	}
+import (
+"log"
```
Comment (byo) [inline src/master/jumpstart.md:345]: Some indentation issues here too

**Review by byo — APPROVED** [review]
Some minor comments about MR linter warnings and code indentation. Apart from that LGTM

File: src/master/develop/connection.md (hunk 4 above)
Comment (mmeloni) [reply src/master/develop/connection.md:15]: I added markdownlint to precommit hooks list, ty

**Pull request merged**